packages/backend/src/open_payments/grant/service.test.ts

```diff
@@ -0,0 +1,110 @@
+import { IocContract } from '@adonisjs/fold'
+import { faker } from '@faker-js/faker'
+import { Knex } from 'knex'
+
+import { GrantOptions, GrantService } from './service'
+import { AccessType, AccessAction } from '../auth/grant'
+import { AuthServer } from '../authServer/model'
+import { initIocContainer } from '../..'
+import { AppServices } from '../../app'
+import { Config } from '../../config/app'
+import { createTestApp, TestContainer } from '../../tests/app'
+import { truncateTables } from '../../tests/tableManager'
+
+describe('Grant Service', (): void => {
+  let deps: IocContract<AppServices>
+  let appContainer: TestContainer
+  let grantService: GrantService
+  let knex: Knex
+
+  beforeAll(async (): Promise<void> => {
+    deps = await initIocContainer(Config)
+    appContainer = await createTestApp(deps)
+    knex = await deps.use('knex')
+  })
+
+  beforeEach(async (): Promise<void> => {
+    grantService = await deps.use('grantService')
+  })
+
+  afterEach(async (): Promise<void> => {
+    await truncateTables(knex)
+  })
+
+  afterAll(async (): Promise<void> => {
+    await appContainer.shutdown()
+  })
+
+  describe('Create and Get Grant', (): void => {
+    test.each`
+      newAuthServer
+      ${false}
+      ${true}
+    `(
+      'Grant can be created and fetched (new auth server: $newAuthServer)',
```

Comment:
imo might be easier to split into two separate tests to avoid the `if (newAuthServer)` checks